packages/gatsby/src/query/file-parser.ts

```diff
@@ -522,15 +564,15 @@ export default class FileParser {
 
   async parseFiles(
     files: Array<string>,
-    addError
-  ): Promise<Array<DocumentNode>> {
-    const documents = []
+    addError: AddParseErrorFunction
+  ): Promise<Array<IGraphQLDocumentInFile>> {
+    const documents: Array<IGraphQLDocumentInFile> = []
 
     return Promise.all(
       files.map(file =>
-        this.parseFile(file, addError).then(docs => {
+        this.parseFile(file, addError).then(docs =>
           documents.push(...(docs || []))
```

Comment:
```suggestion
        this.parseFile(file, addError).then((docs = []) =>
          documents.push(...docs)
```

Reply:
@ascorbic I am afraid this only works if `parseFile` would return `undefined`, but since it returns a nullable value, the compiler is complaining.
```
TS2488: Type 'IGraphQLDocumentInFile[] | null' must have a '[Symbol.iterator]()' method that returns an iterator.
```